target/ppc/fpu_helper.c

```diff
@@ -3368,84 +3368,199 @@ static inline target_ulong compute_fp_flags(float_status *excp_ptr)
+                               bool sat, void* env)
+{
+    int64_t result = ((int64_t)a * (int64_t)b);
+    result += (int64_t)c;
+    if (env && result > ((1LL << 31)-1)) {
+        set_vscr_sat(env);
+        return INT_MAX;
+    } else if (env && result < -(1LL << 31)) {
+        set_vscr_sat(env);
+        return INT_MIN;
+    } else {
+        return (int32_t) result;
+    }
+}
+
+static int32_t int32_t_add (int32_t a, int32_t b, void* env)
+{
+    int64_t result = ((int64_t)a);
+    result += (int64_t)b;
+    if (env && result > ((1LL << 31)-1)) {
+        set_vscr_sat(env);
+        return INT_MAX;
+    } else if (env && result < -(1LL << 31)) {
+        set_vscr_sat(env);
+        return INT_MIN;
+    } else {
+        return (int32_t)result;
+    }
+}
+
+/* CODE IN FLOAT SESSION */
+
+static int32_t int32_t_sub (int32_t a, int32_t b, void* env)
+{
+    g_assert_not_reached();
+    return 0;
+}
+
+static int32_t int32_t_mul (int32_t a, int32_t b, void* env)
+{
+    g_assert_not_reached();
+    return 0;
+}
+
+static inline target_ulong int32_t_check_muladd(int32_t a, int32_t b,
+                                                int32_t c, CPUPPCState *env)
+{
+    g_assert_not_reached();
+    return 0;
+}
+static inline target_ulong int32_t_check_add(int32_t a, int32_t b,
+                                             CPUPPCState *env)
+{
+    g_assert_not_reached();
+    return 0;
+}
+static inline target_ulong int32_t_compute_snan(int32_t a, CPUPPCState *env)
+{
+    g_assert_not_reached();
+    return 0;
+}
+
+#define GET_VsrN(A, I) ((a->VsrB((I)/2) >> ((((I)+1) % 2)*4)) & 0xF)
+#define GET_VsrB(A, I) A->VsrB(I)
+#define GET_VsrH(A, I) A->VsrH(I)
+#define GET_VsrSN(A, I) (GET_VsrN(A,I) | (((GET_VsrN(A,I) >> 3) & 1) ? (-1)^0xF : 0))
+#define GET_VsrSB(A, I) A->VsrSB(I)
+#define GET_VsrSH(A, I) A->VsrSH(I)
+
+/*
+ * int conversion with 2 parameters to mantain same numbers of
+ * parameters as the floating point ones
+ */
+#define int8_t_to_int32_t(A, B) ((int32_t)A)
+#define int16_t_to_int32_t(A, B) ((int32_t)A)
+#define int32_to_int32_t(A, B) A
 
 #define float16_to_float32(A, PTR) float16_to_float32(A, true, PTR)
 
 #define GET_VSR(VSR, A, I, SRC_T, TARGET_T)                             \
     SRC_T##_to_##TARGET_T(GET_##VSR(A, I), excp_ptr)
 
-#define VSXGER(NAME, ORIG_T, OR_EL, TARGET_T, TG_EL, RANK)              \
+#define VSXGER(NAME, ORIG_T, OR_EL, TARGET_T, TG_EL, RANK, FLOAT)       \
     void NAME(CPUPPCState *env, uint32_t a_r, uint32_t b_r,             \
               uint32_t  at_r, uint32_t mask, uint32_t packed_flags)     \
     {                                                                   \
         uint8_t pmsk, xmsk, ymsk;                                       \
         uint8_t pmsk_bit, xmsk_bit, ymsk_bit, ymax;                     \
-        ppc_vsr_t *a, *b, *at;                                          \
+        ppc_vsr_t *a, *b __attribute__((unused)), *at;                  \
```

Comment:
This ```__atribute__((unused))``` shouldn't be necessary, but I found compiling problems without it (even though b is used), if this merge go forward it should be necessary to look into this.

Reply:
The compilation error is probably because of the typo in GET_VsrN I mentioned before.